get_embedding(): remove vertices that do not exist

Ordoviz opened the pull request:
A graph embedding is a Python dictionary `d` where `d[v]` is the list of neighbors of vertex v. When vertices get deleted, the embedding is not updated so [`get_embedding()`](https://doc-develop--sagemath.netlify.app/html/en/reference/graphs/sage/graphs/generic_graph.html#sage.graphs.generic_graph.GenericGraph.get_embedding) needs to remove all vertices in `d[v]` that no longer exist. This commit fixes #40054 where not enough vertices in `d[v`] were removed.

This resulted in `LookupError` being thrown by `get_embedding()` if the graph is immutable.

## :memo: Checklist

- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

Ordoviz committed:
get_embedding: remove vertices that do not exist

An embedding is a Python dictionary d where d[v] is the list of neighbors
of vertex v. When vertices get deleted, the embedding is not updated
immediately so get_embedding() needs to remove all vertices in d[v] that
no longer exist. This commit fixes #40054 where not enough vertices in
d[v] were removed.

dcoudert reviewed (commented):
this looks ok, but why not fixing method `subgraph` which tries to update the embedding ?

Ordoviz committed:
subgraph: do not modify embedding

subgraph() can be lazy because get_embedding() will remove
the old vertices of the supergraph from the embedding.

Ordoviz committed:
use public getter method for graph _embedding

Ordoviz committed:
faces: validate embedding argument

This generates a nicer error if the caller provided an invalid embedding.
genus() call this function with `embedding=self._embedding` but
`self._embedding` might no longer be valid after subgraph operations.

dcoudert reviewed (commented):
Have you also check that the behavior is ok when adding a vertex to the graph ?

dcoudert reviewed (approved):
LGTM.
